### PR TITLE
Switch from Fabric3 to Invoke

### PR DESCRIPTION
**Pull request by bensteinberg**
Since we only use Fabric to run things locally, we can switch to Invoke, removing a few dependencies. Thanks to @lizadaly for the tip.

When writing new Invoke tasks, the two things to remember are including the context as the first argument, and that arguments are not necessarily strings, as they are in Fabric3 -- see https://docs.pyinvoke.org/en/stable/concepts/invoking-tasks.html#type-casting

Before merging this, I'll need to update the `find_pending_cancellation_requests` cron job in our deployment config.

### CODE REVIEW

**Comment by rebeccacremona** [comment]
Oh neat! This is great! `invoke` makes these command lines sound even more like spells, which I am a huge fan of.

Question: so presumably we'll want to do the same thing across all our projects... where `tasks.py` is currently used for Celery. Is it worth figuring out here/now how to use a different file name? Maybe we'll want to rename both going forward, and have `celery_tasks.py` and `invoke_tasks.py`...

**Review by rebeccacremona — APPROVED** [review]

**Comment by bensteinberg** [comment]
Yeah, I was thinking about that. I'm trying to change invoke's `tasks.collection_name` for discovery, stay tuned.

**Comment by bensteinberg** [comment]
Unfortunately, Invoke only gets to "project-level" config (i.e. `invoke.yml` in the current working directory) after it's loaded tasks, which means that to set `collection_name` to something other than `tasks`, you need to do so in e.g. `/etc/invoke.yml` or `~/.invoke.yml`. There's [some discussion](https://github.com/pyinvoke/invoke/issues/537) of being able to configure this in a `[tool.invoke.tasks]` section of `pyproject.toml`, but it's not implemented at the moment.

In other words, I think I need to leave this code in `tasks.py`, and when we make this change in a project that uses Celery, we'll rename `tasks.py` (or `tasks/`) and direct Celery accordingly with something like `app.autodiscover_tasks(related_name='celery_tasks')` in `celery.py`.

**Comment by bensteinberg** [comment]
(You can, of course, specify the collection name on the command line (`-c mytasks`), and possibly make an alias that includes the switch, but I don't like that as much. Happy to be convinced, though.)

**Comment by rebeccacremona** [comment]
> There's https://github.com/pyinvoke/invoke/issues/537 of being able to configure this in a [tool.invoke.tasks] section of pyproject.toml, but it's not implemented at the moment.

Bummer!

👍 on the approach of letting invoke have tasks.py and renaming celery, going forward

**Comment by bensteinberg** [comment]
On rereading https://github.com/harvard-lil/h2o/issues/1810 just now, I see

> Another, perhaps more streamlined alternative, is to just use custom Django management commands

which might have made more sense here, though I think it would have been a bigger change. If anyone has a preference for doing this that way, please let me know.

**Comment by rebeccacremona** [comment]
I think it's probably good to have recourse to something that doesn't need Django to be fully set up yet, for things like [setting up the dev db](https://github.com/harvard-lil/perma/blob/develop/init.sh#L4)... but my example, where a shell script calls a fab task that calls multiple Django management commands is maybe not an example of an elegant design 😂

**Comment by lizadaly** [comment]
FWIW the most common pattern I see in use for this is a `Makefile`, and when complexity exceeds that, just a bash script.
